From c145bfd25c8249fcb434600c986de89f8a19cbf1 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Mon, 3 Feb 2025 19:10:10 +0100
Subject: [PATCH 137/140] ARM64: dts: st: add power domain support in rproc for
 stm32mp25

Add "default" and "sleep" PM domains in rproc node for CM33 low power
in SRAM.

These 2 domains allow to correctly handle CPUIDLE and S2IDLE modes;
- when keep-power-in-suspend is activated in board. the driver selects
  the power domain "sleep" considering that the CM33 FW is running in
  RETRAM or MCU SRAMs,
- when keep-power-in-suspend is deactivated in board. the driver
  selects the power domain "sleep" considering that the CM33 FW is
  running in DDR.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/linux.git 8dfb0a670c4253ab2f76af54dfa7606799cf0684 ARM64: dts: st: add power domain support in rproc for stm32mp25]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I160a687e917a148b4200a6d20f2ec778031afef7
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/434119
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 2eff7d8e24f12..7b6f714d86113 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -2348,9 +2348,10 @@ m33_rproc: m33@0 {
 			interrupts = <GIC_SPI 4 IRQ_TYPE_LEVEL_HIGH>;
 			nvmem-cells = <&rsc_tbl_addr>, <&rsc_tbl_size>;
 			nvmem-cell-names = "rsc-tbl-addr", "rsc-tbl-size";
+			power-domains = <&CLUSTER_PD>, <&RET_PD>;
+			power-domain-names = "default", "sleep";
 
 			status = "disabled";
 		};
 	};
-
 };
-- 
2.51.0

